### PR TITLE
[Refactor](inverted index) refactor inverted index file writer for v1/v2 index write

### PR DESCRIPTION
**Pull request by airborne12**
## Proposed changes

The original write_v1 and write_v2 methods in the inverted index file writer were overly long and difficult to maintain. This PR refactors these methods to improve code maintainability and readability.

### CODE REVIEW

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run p1

**Comment by airborne12** [comment]
run cloud_p1

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp [hunk 1]
```diff
@@ -256,120 +256,32 @@ Status InvertedIndexFileWriter::write_v1() {
...
-            InvertedIndexFileInfo_IndexInfo index_info;
-            index_info.set_index_id(index_id);
-            index_info.set_index_suffix(index_suffix);
-            index_info.set_index_file_size(compound_file_size);
-            auto* new_index_info = _file_info.add_index_info();
-            *new_index_info = index_info;
+            add_index_info(index_id, index_suffix, compound_file_size);
+
         } catch (CLuceneError& err) {
             auto index_path = InvertedIndexDescriptor::get_index_file_path_v1(
```
Comment (csun5285) [inline be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp:286]: finalize_output_dir(out_dir);

**Review by csun5285 — COMMENTED** [review]
line 148 and line 163 is duplicated.

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Review by csun5285 — APPROVED** [review]
LGTM

**Review by zzzxl1993 — APPROVED** [review]
LGTM

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Review by eldenmoon — APPROVED** [review]
LGTM

**Pull request merged**